### PR TITLE
HAI-1611 Make it possible to enable/disable features in different environments

### PR DESCRIPTION
**Pull request by markohaarni**
# Description

Took feature flags into use so that certain features can be toggled on or off. Flags are set with environment variables.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Other

# Instructions for testing

Please describe tests how this change or new feature can be tested.

# Checklist:

- [ ] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

### CODE REVIEW

File: src/common/components/featureFlags/FeatureFlagsContext.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export type FeatureFlagsContextProps = {
+  publicHankkeet: boolean;
+  hanke: boolean;
+  accessRights: boolean;
+};
+
+const FeatureFlagsContext = createContext<FeatureFlagsContextProps | undefined>(undefined);
+
+export function FeatureFlagsProvider({ children }: FeatureFlagsProviderProps) {
+  const value: FeatureFlagsContextProps = {
```
Comment (pitkni) [inline src/common/components/featureFlags/FeatureFlagsContext.tsx:16]: Sonar seems to have an issue with this:

The 'value' object passed as the value prop to the Context provider changes every render. To fix this consider wrapping it in a useMemo hook.)

So maybe something like:

```
  const value: FeatureFlagsContextProps = useMemo(
    () => ({
      publicHankkeet: window._env_.REACT_APP_FEATURE_PUBLIC_H === '1',
      hanke: window._env_.REACT_APP_FEATURE_H === '1',
      accessRights: window._env_.REACT_APP_FEATURE_ACCESS_RIGHTS === '1',
    }),
    []
  );
```

Comment (pitkni) [reply src/common/components/featureFlags/FeatureFlagsContext.tsx:16]: I'm not sure if the dependency array should be empty or not.

Comment (markohaarni) [reply src/common/components/featureFlags/FeatureFlagsContext.tsx]: Empty in this case as there are no dependencies.

Comment (markohaarni) [reply src/common/components/featureFlags/FeatureFlagsContext.tsx]: Wrapped the value in useMemo.

**Review by pitkni — APPROVED** [review]

**Pull request merged**